core/src/main/java/com/linecorp/armeria/common/util/AbstractListenable.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.function.Consumer;
+
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * A skeletal {@link Listenable} implementation.
+ */
+public abstract class AbstractListenable<T> implements Listenable<T> {
+    private final Set<Consumer<? super T>> updateListeners = new CopyOnWriteArraySet<>();
+
+    /**
+     * Notify the {@link Endpoint}s changes to listeners.
```

Comment:
- `{@link Endpoint}s changes` -> `new value`?
- `listeners` -> `the listeners added via {@link #addListener(Consumer)}`